packages/generator/src/options/options-parser.ts

```diff
@@ -47,37 +100,43 @@ export function getOptionsWithoutDefaults<
  */
 export function parseOptions<
   CliOptionsT extends Record<string, Option>,
-  GeneratorOptionsT extends Record<string, any>
+  GeneratorOptionsT extends Partial<Record<keyof CliOptionsT, any>>
 >(
   options: CliOptionsT,
   userOptions: GeneratorOptionsT
-): ParsedOptions<GeneratorOptionsT, CliOptionsT> {
+): ParsedOptions<CliOptionsT> {
   return new OptionsParser(options, userOptions).parseOptions();
 }
 
 class OptionsParser<
   CliOptionsT extends Record<string, Option>,
-  GeneratorOptionsT extends Record<string, any>
+  GeneratorOptionsT extends Record<keyof CliOptionsT, any>
 > {
   constructor(
     private options: CliOptionsT,
     private userOptions: GeneratorOptionsT
   ) {}
 
-  parseOptions(): ParsedOptions<GeneratorOptionsT, CliOptionsT> {
+  parseOptions(): ParsedOptions<CliOptionsT> {
```

Comment:
[req] I think it would be good to have a way to get the options used by the `generateWithParsedOptions` so I would perhaps add here a log statement showing the final option being passed. Doing so we can ask for that output on support issues. This should be a low log level `debug`  programmatic use case OR if the `verbose` option is activated in the cli use case. - If we have something like this already please ignore this comment.

Reply:
Done.